src/main/java/io/confluent/connect/jdbc/source/TableQuerier.java

```diff
@@ -81,8 +83,21 @@ public boolean querying() {
   public void maybeStartQuery(Connection db) throws SQLException {
     if (resultSet == null) {
       stmt = getOrCreatePreparedStatement(db);
+
       resultSet = executeQuery();
-      schema = DataConverter.convertSchema(name, resultSet.getMetaData(), mapNumerics);
+      ResultSetMetaData qr = resultSet.getMetaData();
```

Comment:
Does this query get the metadata from the DB regardless of if the ResultSet has non-0 results?

Reply:
Looks like if the result set contains 0 rows, the metadata is still all there

Reply:
Interesting. Well, if we find out we have performance issues we can look at it, otherwise I suppose it's part of the standard to return the DB metadata with the content.